selectel/mks.go

```diff
@@ -655,3 +656,133 @@ func parseMKSKubeVersionsV1Default(versions []*kubeversion.View) string {
 
 	return defaultVersion
 }
+
+// findQuota finds and returns quota for specified resource from quotas slice.
+func findQuota(quotas []*quotas.Quota, resource string) []quotas.ResourceQuotaEntity {
+	for _, q := range quotas {
+		if q.Name == resource {
+			return q.ResourceQuotasEntities
+		}
+	}
+
+	return nil
+}
+
+func checkQuotasForCluster(projectQuotas []*quotas.Quota, region string, zonal bool) error {
+	var (
+		clusterQuotaChecked bool
+		quota               []quotas.ResourceQuotaEntity
+	)
+
+	if zonal {
+		quota = findQuota(projectQuotas, "mks_cluster_zonal")
+	} else {
+		quota = findQuota(projectQuotas, "mks_cluster_regional")
+	}
+
+	if quota == nil {
+		return errors.New("unable to find zonal or regional k8s cluster quotas")
```

Comment:
Just like in the last commit let's return two different errors here in the case of checking zonal or regional quotas. Also don't forget to update unit tests.